pkg/polaris/polaris_test.go

```diff
@@ -748,3 +751,372 @@ func TestGcpPermissions(t *testing.T) {
 		t.Fatal("invalid number of permissions: 0")
 	}
 }
+
+// TestListSLA verifies that the SDK can list the default SLAs - Gold, Silver
+// and Bronze.
+//
+// To run this test against a Polaris instance the following environment
+// variables needs to be set:
+//   * TEST_INTEGRATION=1
+//   * RUBRIK_POLARIS_SERVICEACCOUNT_FILE=<path-to-polaris-service-account-file>
+func TestListSLA(t *testing.T) {
+	if !boolEnvSet("TEST_INTEGRATION") {
+		t.Skipf("skipping due to env TEST_INTEGRATION not set")
+	}
+
+	ctx := context.Background()
+
+	// Load configuration and create client. Usually resolved using the
+	// environment variable RUBRIK_POLARIS_SERVICEACCOUNT_FILE.
+	polAccount, err := DefaultServiceAccount(true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	client, err := NewClient(ctx, polAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	slaDomains, err := client.Core().ListSLA(
+		ctx,
+		core.SortByName,
+		core.ASC,
+		[]core.GlobalSLAFilterInput{
+			{
+				Field: core.ObjectType,
+				Text:  "",
+				ObjectTypeList: []core.SLAObjectType{
+					core.KuprObjectType,
+				},
+			},
+		},
+		core.Default,
+		[]core.ContextFilterInputField{{"", ""}},
+		false,
+		false,
+		false,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	basicCheck := make(map[string]bool)
+	for _, slaDomain := range slaDomains {
+		if slaDomain.Name == "Gold" || slaDomain.Name == "Silver" || slaDomain.Name == "Bronze" {
+			basicCheck[slaDomain.Name] = true
+		}
+	}
+	if !basicCheck["Gold"] || !basicCheck["Silver"] || !basicCheck["Bronze"] {
+		t.Errorf("failed to list default SLAs: %v", basicCheck)
+	}
+}
+
+// TestListSLA verifies that the SDK can list the default SLAs - Gold, Silver
+// and Bronze.
+//
+// To run this test against a Polaris instance the following environment
+// variables needs to be set:
+//   * TEST_INTEGRATION=1
+//   * RUBRIK_POLARIS_SERVICEACCOUNT_FILE=<path-to-polaris-service-account-file>
+func TestK8sListSLA(t *testing.T) {
+	//if !boolEnvSet("TEST_INTEGRATION") {
+	//	t.Skipf("skipping due to env TEST_INTEGRATION not set")
+	//}
+
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|2YDsG5FIYWQ7OW8AiRqA2tPwQEwhHpfU",
+		ClientSecret:   "ZLeoEHA6dvDGciAO1bFIIWJmdjlLLawFa6WC8IdpBfgphecsUQYB4pAOKovQbt4e",
```

Comment:
:exclamation:**Cycode**: Secret of type: 'Base64 Entropy' was found.
Severity: Medium <!-- 6cd11a0a-2909-4a77-801a-919ca0db48a2 -->